src/Handlebars/Context.php

```diff
@@ -32,21 +37,42 @@ class Context
      */
     protected $index = [];
 
+    /**
+     * @var array dataStack stack for data within sections
+     */
+    protected $dataStack = [];
+
     /**
      * @var array key stack for objects
      */
     protected $key = [];
 
+    /**
+     * @var bool enableDataVariables true if @data variables should be used.
+     */
+    protected $enableDataVariables = false;
+
     /**
      * Mustache rendering Context constructor.
      *
      * @param mixed $context Default rendering context (default: null)
+     * @param array $options Options for the context. It may contain the following: (default: empty array)
+     *                       enableDataVariables => Boolean, Enables @data variables (default: false)
      */
-    public function __construct($context = null)
+    public function __construct($context = null, $options = [])
     {
         if ($context !== null) {
             $this->stack = [$context];
         }
+
+        if (isset($options[Handlebars::OPTION_ENABLE_DATA_VARIABLES])) {
+            if (!is_bool($options[Handlebars::OPTION_ENABLE_DATA_VARIABLES])) {
```

Comment:
Do you mind combining these into a single conditional and then setting ```$this->enableDataVariables = $options[Handlebars::OPTION_ENABLE_DATA_VARIABLES];``` after the conditional check since it will throw an Exception if it's true?

Reply:
I'm not sure I understand.

```php
if (isset($options[...]) && !is_bool($options[...])) {
    throw ... 
}
$this->enableDataVariables = $options[...];
```
Is not equivalent as it would throw an exception when the option is not specified in the constructor (like `new Context()`).

If this library were using PHP >7 we could use the coalesce operator but it's currently at PHP support 5.4.

Reply:
Oh right dis-regard, I misread the closing of the if statements.